lib/checks/aria/valid-scrollable-semantics.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * A map from HTML tag names to a boolean which reflects whether it is
+ * appropriate for scrollable elements found in the focus order.
+ */
+const VALID_TAG_NAMES_FOR_SCROLLABLE_REGIONS = {
+  ARTICLE: true,
+  ASIDE: true,
+  NAV: true,
+  SECTION: true
+};
+
+/**
+ * A map from each landmark role to a boolean which reflects whether it is
+ * appropriate for scrollable elements found in the focus order.
+ */
+const VALID_ROLES_FOR_SCROLLABLE_REGIONS = {
+  banner: false,
+  complementary: true,
+  contentinfo: true,
+  form: true,
+  main: true,
+  navigation: true,
+  region: true,
+  search: false
+};
+
+/**
+ * @param {HTMLElement} node
+ * @return {Boolean} Whether the element has a tag appropriate for a scrollable
+ *     region.
+ */
+function validScrollableTagName(node) {
```

Comment:
I don't understand the purpose of this. The only part of the selector that lets these elements in is if they match `[role=heading]`. I don't think `<nav role="heading" tabindex="1">` should pass this rule, and I don't see any other path leading to this function passing anything either.

Reply:
The example nav[role=heading][tabindex=1] should presumably fail a different test, since navs should not be allowed to have role=heading (or any other ARIA role).

My intuition is to make this check available for use in future rules, rules which may not have the same selector as our focus-order-semantics rule.

I'd argue that this check should stand on its own feet, and should PASS elements that have one of these tags as plausibly being scrollable regions. This seems more maintainable and safer with respect to false positives, but I happily defer to your experience with axe. What do you think?

Reply:
I'm fine with that. Just making sure I understand what's going on. Can you add a comment in to explain this?

Reply:
Done.

Reply:
let me know if the comment I left wasn't what you were suggesting